ami/utils.go

```diff
@@ -90,9 +139,23 @@ func requestList(ctx context.Context, client Client, action, id, event, complete
 
 	response := make([]Response, 0)
 	for {
-		rsp, err := read(ctx, client)
-		if err != nil {
-			return nil, err
+		var rsp Response
+		var err error
+		if useEventBroker {
+			select {
+			case rsp = <-messageChan:
+			case <-ctx.Done():
+				return nil, nil
+			}
+		} else {
+			rsp, err = read(ctx, client)
+			if err != nil {
+				return nil, err
+			}
+		}
+		actionID := rsp.Get("ActionID")
```

Comment:
I agree that *ActionID* should be the same of included in the request and the verification **must** be included as fix.

Reply:
i will create an new pull request with this fix